packages/vaadin-template-renderer/src/vaadin-template-renderer-templatizer.js

```diff
@@ -12,49 +12,47 @@ export class Templatizer extends PolymerElement {
     return 'vaadin-template-renderer-templatizer';
   }
 
-  static get properties() {
-    return {
-      __template: Object,
-      __TemplateClass: Object,
-      __templateInstance: Object
-    };
-  }
+  constructor() {
+    super();
 
-  /**
-   * @public
-   */
-  render(element) {
-    // TODO: Support creating multiple instances of the template
-    if (!this.__templateInstance) {
-      this.__createTemplateInstance();
+    this.__template = null;
+    this.__TemplateClass = null;
+    this.__templateInstances = new Set();
+  }
 
-      element.__templateInstance = this.__templateInstance;
-      element.replaceChildren(this.__templateInstance.root);
+  render(element, properties) {
+    // If the template instance exists and has been instantiated by this templatizer,
+    // it only re-renders the instance with the new properties.
+    if (this.__templateInstances.has(element.__templateInstance)) {
+      element.__templateInstance.setProperties(properties);
+      return;
     }
+
+    // Otherwise, it instantiates a new template instance
+    // with the given properties and then renders the result to the element
+    const templateInstance = this.__createTemplateInstance(properties);
+    element.__templateInstance = templateInstance;
+    element.replaceChildren(templateInstance.root);
```

Comment:
Looks like [`replaceChildren`](https://caniuse.com/mdn-api_element_replacechildren) is only supported since Safari 14 so we probably want to use some older API.

Reply:
Fixed.